src/uucore/src/lib/mods/clap_localization.rs

```diff
@@ -0,0 +1,183 @@
+                        colorize(&error_word, colors::RED),
+                        colorize(&tip_word, colors::GREEN),
+                    )
+                } else {
+                    (arg_str.clone(), error_word.clone(), tip_word.clone())
+                };
+
+                // Print main error message
+                let error_msg = translate!(
+                    "clap-error-unexpected-argument",
+                    "arg" => colored_arg.clone(),
+                    "error_word" => colored_error_word
+                );
+                eprintln!("{error_msg}");
+                eprintln!();
+
+                // Show suggestion or generic tip
+                let suggestion = err.get(ContextKind::SuggestedArg);
+                if let Some(suggested_arg) = suggestion {
+                    let colored_suggestion = if colors_enabled {
+                        colorize(&suggested_arg.to_string(), colors::GREEN)
+                    } else {
+                        suggested_arg.to_string()
+                    };
+                    let suggestion_msg = translate!(
+                        "clap-error-similar-argument",
+                        "tip_word" => colored_tip_word,
+                        "suggestion" => colored_suggestion
+                    );
+                    eprintln!("  {suggestion_msg}");
+                } else {
+                    let colored_tip_command = if colors_enabled {
+                        colorize(&format!("-- {arg_str}"), colors::GREEN)
+                    } else {
+                        format!("-- {arg_str}")
+                    };
+                    let tip_msg = translate!(
+                        "clap-error-pass-as-value",
+                        "arg" => colored_arg,
+                        "tip_word" => colored_tip_word,
+                        "tip_command" => colored_tip_command
+                    );
+                    eprintln!("  {tip_msg}");
+                }
+
+                // Show usage and help
+                eprintln!();
+                let usage_label = translate!("common-usage");
+                let usage_pattern = translate!(&format!("{util_name}-usage"));
+                eprintln!("{usage_label}: {usage_pattern}");
+                eprintln!();
+
+                let help_msg = translate!("clap-error-help-suggestion", "command" => util_name);
+                eprintln!("{help_msg}");
+
+                std::process::exit(exit_code);
+            } else {
+                // Generic fallback case
+                let rendered = err.render();
+                let rendered_str = rendered.to_string();
+                let colors_enabled = rendered_str.contains("\x1b[");
+
+                let colored_error_word = if colors_enabled {
+                    colorize(&translate!("common-error"), colors::RED)
+                } else {
+                    translate!("common-error")
+                };
```

Comment:
You use this structure with `if colors_enabled` multiple times, thus it might make sense to move it to a helper closure.